src/jit/importer.cpp

```diff
@@ -7655,6 +7663,7 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             if ((call->gtFlags & GTF_CALL_INLINE_CANDIDATE) != 0)
             {
                 assert(opts.OptEnabled(CLFLG_INLINING));
+                assert(!(call->gtFlags & GTF_CALL_M_FAT_POINTER_CHECK)); // We should not try to inline calli.
```

Comment:
> assert(!(call->gtFlags & GTF_CALL_M_FAT_POINTER_CHECK)); [](start = 15, length = 57)

The check should be on gtCallMoreFlags and we should abstract flag check in a routine.

This comment applies to all places where we are setting and checking GTF_CALL_M_FAT_POINTER_CHECK on call->gtFlags.